Nix/Eval.hs

```diff
@@ -16,16 +16,17 @@ import           Prelude hiding (mapM, sequence)
 buildArgument :: Formals NValue -> NValue -> NValue
 buildArgument paramSpec arg = either error (Fix . NVSet) $ case paramSpec of
     FormalName name -> return $ Map.singleton name arg
-    FormalSet s -> lookupParamSet s
-    FormalLeftAt name s -> Map.insert name arg <$> lookupParamSet s
-    FormalRightAt s name -> Map.insert name arg <$> lookupParamSet s
+    FormalSet s Nothing -> lookupParamSet s
+    FormalSet s (Just name) -> Map.insert name arg <$> lookupParamSet s
```

Comment:
`nix-instantiate --parse` makes no distinction between `@`s at left or right, so I figured this was unnecessary information to retain. Let me know if you feel otherwise.